kratos/utilities/read_materials_utility.cpp

```diff
@@ -0,0 +1,265 @@
+            const Variable<double>& variable = KratosComponents<Variable<double>>().Get(variable_name);
+            if (value.IsDouble()) {
+                p_prop->SetValue(variable, value.GetDouble());
+            } else if (value.IsInt()) {
+                p_prop->SetValue(variable, static_cast<double>(value.GetInt()));
+            } else {
+                KRATOS_ERROR << "Check you write the value in a correct format: " << value << std::endl;
+            }
+        } else if(KratosComponents<Variable<bool> >::Has(variable_name)) {
+            const Variable<bool>& variable = KratosComponents<Variable<bool>>().Get(variable_name);
+            if (value.IsBool()) {
+                p_prop->SetValue(variable, value.GetBool());
+            } else if (value.IsInt()) {
+                p_prop->SetValue(variable, static_cast<bool>(value.GetInt()));
+            } else if (value.IsDouble()) {
+                p_prop->SetValue(variable, static_cast<bool>(value.GetDouble()));
+            } else {
+                KRATOS_ERROR << "Check you write the value in a correct format: " << value << std::endl;
+            }
+        } else if(KratosComponents<Variable<int> >::Has(variable_name)) {
+            const Variable<int>& variable = KratosComponents<Variable<int>>().Get(variable_name);
+            if (value.IsInt()) {
+                p_prop->SetValue(variable, value.GetInt());
+            } else if (value.IsDouble()) {
+                p_prop->SetValue(variable, static_cast<int>(value.GetDouble()));
+            } else {
+                KRATOS_ERROR << "Check you write the value in a correct format: " << value << std::endl;
+            }
+        } else if(KratosComponents<Variable<array_1d<double, 3> > >::Has(variable_name)) {
+            const Variable<array_1d<double, 3>>& variable = KratosComponents<Variable<array_1d<double, 3>>>().Get(variable_name);
+            if (value.IsVector()) {
+                array_1d<double, 3> temp(3, 0.0);
+                const Vector& value_variable = value.GetVector();
+                const std::size_t iter_number = (3 < value_variable.size()) ? 3 : value_variable.size();
+                for (std::size_t index = 0; index < iter_number; index++)
+                    temp[index] = value_variable[index];
+                p_prop->SetValue(variable, temp);
+            } else {
+                KRATOS_ERROR << "Check you write the value in a correct format: " << value << std::endl;
+            }
+        } else if(KratosComponents<Variable<array_1d<double, 6> > >::Has(variable_name)) {
+            const Variable<array_1d<double, 6>>& variable = KratosComponents<Variable<array_1d<double, 6>>>().Get(variable_name);
+            if (value.IsVector()) {
+                array_1d<double, 6> temp(6, 0.0);
+                const Vector& value_variable = value.GetVector();
+                const std::size_t iter_number = (6 < value_variable.size()) ? 6 : value_variable.size();
```

Comment:
I think you should throw an error if the size does not match

Reply:
Well, that can be disussed

Reply:
I think if the size != 6 it is undefined behavior that should not be allowed

we could even think abt implementing IsArray3d / IsArray6d in the Parameters

Reply:
I agree, but @RiccardoRossi tell me to do conversions in order to give more freedom to the user. I agree to define arrays on the Parameters

Reply:
ok then lets wait for @RiccardoRossi

Reply:
+1 for throwing an error.

Reply:
+1 for the error. I could agree about having IsArray3, as it is a common usecase, but i would definitely not want IsArray6 ...